Shared/Supporting Files/Views/FavoritesView.swift

```diff
@@ -0,0 +1,153 @@
+    struct AddFavoriteView: View {
+        /// All samples retrieved from the Samples directory.
+        let samples: [Sample]
+        
+        /// The action to dismiss the sheet.
+        @Environment(\.dismiss) private var dismiss: DismissAction
+        
+        /// The names of the favorited samples loaded from user defaults.
+        @AppStorage(UserDefaults.favoritedSamplesKey) private var favoritedNames: [String] = []
+        
+        /// The search query in the search bar.
+        @State private var query = ""
+        
+        /// The samples displayed in the search list.
+        var displayedSamples: [Sample] {
+            if query.isEmpty {
+                return samples
+            } else {
+                return samples.filter { $0.name.contains(query) }
+            }
+        }
+        
+        var body: some View {
+            NavigationView {
+                List {
+                    ForEach(displayedSamples, id: \.name) { sample in
+                        Button {
+                            if !favoritedNames.contains(sample.name) {
+                                favoritedNames.append(sample.name)
+                            }
+                            dismiss()
+                        } label: {
+                            HStack {
+                                Text(sample.name)
+                                Spacer()
+                            }
+                            .contentShape(Rectangle())
+                        }
+                        .buttonStyle(.plain)
+                    }
+                }
+                .listStyle(.inset)
+                .navigationBarTitleDisplayMode(.inline)
+                .toolbar {
+                    ToolbarItem(placement: .principal) {
+                        Text("Choose a sample to add to Favorites")
+                            .font(.subheadline)
+                    }
+                    ToolbarItem(placement: .navigationBarTrailing) {
+                        Button("Done") {
+                            dismiss()
+                        }
+                    }
+                }
+            }
+            .searchable(text: $query, placement: .navigationBarDrawer(displayMode: .always))
+        }
+    }
+}
+
+// MARK: User Defaults
+
+extension UserDefaults {
+    /// The key to read and write favorited sample names to the user defaults.
+    static var favoritedSamplesKey: String {
+        "favoritedSamples"
+    }
+}
+
+/// An extension allowing an array to be used with the app storage property wrapper.
+extension Array: RawRepresentable where Element: Codable {
+    /// Creates a new array from a given raw value.
+    /// - Parameter rawValue: The raw value of the array to create.
+    public init?(rawValue: String) {
+        guard let data = rawValue.data(using: .utf8),
+              let result = try? JSONDecoder().decode([Element].self, from: data)
+        else { return nil }
```

Comment:
I think there is some disparity here - the default encoded value is an empty array JSON, while the default decoded value is `nil`. What about both using empty array as default?